aptos-move/block-executor/src/view.rs

```diff
@@ -1174,6 +1230,39 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TDelayedFie
             },
         }
     }
+
+    // TODO - update comment.
+    // For each resource that satisfies the following conditions,
+    //     1. Resource is in read set
+    //     2. Resource is not in write set
+    // replace the delayed field identifiers in the resource with corresponding values.
+    // If any of the delayed field identifiers in the resource are part of delayed_field_write_set,
+    // then include the resource in the write set.
+    fn get_reads_needing_exchange(
+        &self,
+        delayed_write_set_keys: &HashSet<Self::Identifier>,
+        skip: &HashSet<Self::ResourceKey>,
```

Comment:
I don't see any `skip` usage being non-empty?

Reply:
Skip is for the writeset, which we don't have at this point.  I have a TODO for that

Reply:
@igor-aptos what does this mean? cc @gelash

Reply:
reads_needing_exchange must not contain any element from the write set. (or we will double charge the gas)

currently we first compute all reads, and then filter out those that are in the writeset. TODO is to see if we can get the set of keys in the write set, and then pass it as skip here, to compute the correct list on the first try (and reduce wasted time on checking identifiers for resources that are in the write set.